Apps/phipqualityscore/ipqualityscore_connector.py

```diff
@@ -0,0 +1,397 @@
+                error_code = ERR_CODE_MSG
+                error_msg = ERR_MSG_UNAVAILABLE
+        except:
+            error_code = ERR_CODE_MSG
+            error_msg = ERR_MSG_UNAVAILABLE
+
+        try:
+            if error_code in ERR_CODE_MSG:
+                error_text = "Error Message: {0}".format(error_msg)
+            else:
+                error_text = "Error Code: {0}. Error Message: {1}".format(error_code, error_msg)
+        except:
+            self.debug_print(PARSE_ERR_MSG)
+            error_text = PARSE_ERR_MSG
+
+        return error_text
+
+    def _validate_integer(self, action_result, parameter, key):
+        if parameter is not None:
+            try:
+                if not float(parameter).is_integer():
+                    return action_result.set_status(phantom.APP_ERROR, VALID_INTEGER_MSG.format(key)), None
+
+                parameter = int(parameter)
+            except:
+                return action_result.set_status(phantom.APP_ERROR, VALID_INTEGER_MSG.format(key)), None
+
+            if parameter < 0:
+                return action_result.set_status(phantom.APP_ERROR, NON_NEGATIVE_INTEGER_MSG.format(key)), None
+
+        return phantom.APP_SUCCESS, parameter
+
+    def test_asset_connectivity(self, param):
+        config = self.get_config()
+        app_key = config['apikey']
+        self.save_progress(IPQUALITYSCORE_MSG_CONNECTING)
+        try:
+            response = requests.get(
+                IPQUALITYSCORE_API_TEST.format(apikey=app_key))
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            self.debug_print('test_asset_connectivity: {}'.format(err))
+            err_msg = '{}. {}. Error Occurred: {}'.format(IPQUALITYSCORE_ERR_CONNECTIVITY_TEST, IPQUALITYSCORE_MSG_CHECK_CONNECTIVITY, err)
+            return self.set_status(phantom.APP_ERROR, err_msg)
+
+        if response.status_code == 509:
+            self.save_progress(IPQUALITYSCORE_SERVER_ERROR_RATE_LIMIT)
+            self.save_progress(IPQUALITYSCORE_ERR_CONNECTIVITY_TEST)
+            return self.set_status(phantom.APP_ERROR)
+        if response.status_code != 200:
+            self.save_progress('{}. {}'.format(IPQUALITYSCORE_SERVER_RETURNED_ERROR_CODE.
+                        format(code=response.status_code), IPQUALITYSCORE_MSG_CHECK_CONNECTIVITY))
+            self.save_progress(IPQUALITYSCORE_ERR_CONNECTIVITY_TEST)
+            return self.set_status(phantom.APP_ERROR)
+
+        try:
+            result = response.json()
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            self.debug_print('Response from server is not a valid JSON {}'.format(err))
+            self.save_progress('Response from server is not a valid JSON')
+            self.save_progress(IPQUALITYSCORE_ERR_CONNECTIVITY_TEST)
+            return self.set_status(phantom.APP_ERROR)
+
+        if 'success' in result and result['success'] is True:
```

Comment:
```suggestion
        if result.get('success'):
```

same for all other instances of the same in this file since there are multiple

Reply:
Fixed